pkg/auth/oauth/external/gitlab/gitlab.go

```diff
@@ -0,0 +1,111 @@
+package gitlab
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/RangelReale/osincli"
+	"github.com/golang/glog"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/auth/oauth/external"
+)
+
+const (
+	gitlabAuthorizePath = "oauth/authorize"
+	gitlabTokenPath     = "oauth/token"
+	gitlabUserApiPath   = "api/v3/user"
+	gitlabOAuthScope    = "api"
```

Comment:
I had a hard time finding out gitlab scope info... is this the minimal scope that can be used to get user info? any chance of pointing to reference documentation on the scope?